diff --git a/lib/fantom/esm/util.js b/lib/fantom/esm/util.js
index 36c1b9c..19cf250 100644
--- a/lib/fantom/esm/util.js
+++ b/lib/fantom/esm/util.js
@@ -714,6 +714,54 @@ class FileLocErr extends sys.Err {
 
 }
 
+class LockFile extends sys.Obj {
+  constructor() {
+    super();
+    const this$ = this;
+  }
+
+  typeof() { return LockFile.type$; }
+
+  #file = null;
+
+  file() { return this.#file; }
+
+  __file(it) { if (it === undefined) return this.#file; else this.#file = it; }
+
+  /**
+   * Release the lock if we are holding one
+   */
+  unlock() {
+    return this
+  }
+  /**
+   * Backing file we use to lock/acquire
+   */
+  file() {
+    return this.#file;
+  }
+  /**
+   * Acquire the lock or raise CannotAcquireLockFileErr
+   */
+  lock() {
+    return this
+  }
+  /**
+   * Construct with given file
+   */
+  static make(file, args) {
+    const $self = new LockFile();
+    LockFile.make$($self, file, args);
+    return $self;
+  }
+
+  static make$($self, file, args) {
+    $self.#file = file;
+    return;
+  }
+
+}
+
 class Macro extends sys.Obj {
   constructor() {
     super();
